paddle/fluid/framework/prune.cc

```diff
@@ -81,19 +83,47 @@ bool HasFalseTarget(const proto::OpDesc& op_desc) {
 }
 
 int GetSubBlockIndex(const proto::OpDesc& op_desc) {
+  // the block index >= 0, so -1 is used to indicate "NotFound".
   for (auto& attr : op_desc.attrs()) {
     if (attr.type() == proto::AttrType::BLOCK) {
-      PADDLE_ENFORCE(attr.has_block_idx());
+      PADDLE_ENFORCE_EQ(
+          attr.has_block_idx(), true,
+          platform::errors::NotFound("Attribute sub_block is empty"));
       return attr.block_idx();
     }
   }
   return -1;
 }
 
+void SetSubBlockIndex(proto::OpDesc* op_desc, int sub_idx) {
+  for (auto& attr : *op_desc->mutable_attrs()) {
+    if (attr.type() == proto::AttrType::BLOCK) {
+      PADDLE_ENFORCE_EQ(
+          attr.has_block_idx(), true,
+          platform::errors::NotFound("Attribute sub_block is empty"));
```

Comment:
Better to print the Op's name in the error message.

Reply:
Done, thanks.